common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -53,6 +53,43 @@ def get_queryset(self):
         # to Admin users
         return CourseEntitlement.objects.all().select_related('user').select_related('enrollment_course_run')
 
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+
+        username = serializer.data.get('user')
+        user = User.objects.get(username=username)
+        entitlement = CourseEntitlement.objects.get(uuid=serializer.data.get('uuid'), user=user.id, expired_at=None)
+
+        # find all course_runs within the course        
+        course_runs = get_course_runs_for_course(serializer.data.get('course_uuid'))
+
+        # check if the user has audit enrollments for any of the course_runs
+        user_run_enrollments = {CourseEnrollment.get_enrollment(user, CourseKey.from_string(course_run.get('key'))) for course_run in course_runs if CourseEnrollment.get_enrollment(user, CourseKey.from_string(course_run.get('key')))}
+        
+        # filter out unupgradeable user enrollments
+        upgradeable_user_run_enrollments = []
+        if user_run_enrollments:
```

Comment:
I think you can define `upgradeable_user_run_enrollments` doing this?
```python
filter(lambda e: e.upgrade_deadline and e.upgrade_deadline > timezone.now(), user_run_enrollments)
```

Reply:
^ Following up on that, you probably don't need the if user_run_enrollments check, since if its an empty list, the for loop just won't run